### ISSUE TITLE
`bun run <missing.ts>` should exits with non-0

### ISSUE THREAD

**Issue by azu** [oven-sh/bun#4011]
### What version of Bun is running?

0.7.2

### What platform is your computer?

Darwin 22.6.0 arm64 arm

### What steps can reproduce the bug?

```
$ bun run missing.ts
error: missing script "missing.ts"
$ echo $?
0
```

### What is the expected behavior?

Exit status should be non-zero.

Node.js: `1`

```
❯ node missing.js
node:internal/modules/cjs/loader:1080
  throw err;
  ^

Error: Cannot find module '/Users/azu/ghq/github.com/azu/bluenotiondb/missing.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at Module._load (node:internal/modules/cjs/loader:922:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v18.17.0

❯ echo $?
1
```

Deno: `130`

```
❯ deno run missing.ts
error: Module not found "file:///Users/azu/ghq/github.com/azu/bluenotiondb/missing.ts".
❯ echo $?
130
```

### What do you see instead?

`bun run missing.ts` exits with `0`.

### Additional information

This behaviour caused problems where CI would pass even if the wrong file name was specified.

`bun run` show `error: missing script "index.ts"`, but this CI is passed.
<img width="1310" alt="image" src="https://github.com/oven-sh/bun/assets/19714/ed72d97b-1746-4e4d-8023-ac074284d4aa">

**Comment by Jarred-Sumner** [comment]
yes it should, thank you for flagging

**Comment by Jarred-Sumner** [comment]
Fixed by @YashoSharma in #4026
